editor/src/clj/editor/graphics.clj

```diff
@@ -592,6 +653,15 @@
 (def ^:private renderable-data->world-normal-v4 (partial renderable-data->world-direction-v4 :normal-data))
 (def ^:private renderable-data->world-tangent-v3 (partial renderable-data->world-direction-v3 :tangent-data))
 
+(defn- matrix4+attribute->flat-array [^Matrix4d matrix attribute]
+  (let [vector-component-count (vector-type->component-count (:vector-type attribute))
+        matrix-flat-array (math/vecmath->clj (doto (Matrix4d. matrix) (.transpose)))
+        matrix-row-column-count (vtx/vertex-attribute->row-column-count attribute)]
+    (vec (flatten (if matrix-row-column-count
+                    (map #(take matrix-row-column-count %)
+                         (take matrix-row-column-count (partition 4 matrix-flat-array)))
+                    (take vector-component-count matrix-flat-array))))))
```

Comment:
Maybe this can be written more efficiently? I couldn't find a nice way in the matrix class to grab a sub-matrix